src/client/pfs_v2.go

```diff
@@ -40,8 +41,10 @@ func (c APIClient) DeleteFilesV2(repo, commit string, files []string, tag ...str
 // FileOperationClient is not thread safe. Multiple FileOperationClients
 // should be used for concurrent upload.
 type FileOperationClient struct {
```

Comment:
So I think we should probably tweak this design a bit. I think the basic idea of reusing the `FileOperationV2` functionality makes sense, and I think there is probably more we can refactor on the server side to really make this clean (essentially just have the interpreting of the operations / setting up of the unordered writer in helper functions that both rpcs use). The server side changes can come later, but I think we want to go ahead and tweak this client stuff a bit, because the closing stuff is pretty messy. I think we could set this up such that the `FileOperationV2` and `CreateTempFileSet` client wrappers each have their own abstraction which embeds something that functions similarly to the `FileOperationClient`. The key change though would be that the embedded abstraction would just take a client and be able to send `FileOperationRequestV2`s, but it would have no notion of closing the client. The outer abstraction would have a close method attached to it that would handle the closing differences between the two rpcs. Therefore, we could hand back a client for each rpc that could perform all of the operations as well as have the correct closing semantics.